### ISSUE TITLE
Train face and the whole body at the same time

### ISSUE THREAD

**Issue by AkiKagura** [JoePenna/Dreambooth-Stable-Diffusion#93]
If I use facial pictures as training data, when I generate full body portrait, I will get some pictures with distorted face; 
If I use full body pictures as training data, the generated model cannot freely specify hair style and clothing.They all in the style of my training images.(eg If I use person pics in jeans and T-shirts I cannot let him/her wearing dress)
I want to know how to get a training model that can generate pictures which can freely assign character clothing without generating distorted faces?
Thank you really much for helping.🙏

**Comment by AkiKagura** [comment]
My solution is to select an appropriate number of face portraits and upper-body portraits as training pictures. For each type , there should be an appropriate amount of training pictures. The human proportion between pictures should not be too different. In addition, the number of training steps should not be too large(I found 3000 on 37 pics is enough).

**Comment by AlexanderKozhevin** [comment]
@AkiKagura how many images did you use for training?

**Comment by AkiKagura** [comment]
@AlexanderKozhevin about 37pics, 3000steps, on RTX3090, 1.5h

**Comment by AkiKagura** [comment]
And the amount of training pics does not seem to affect the training time

**Comment by yuxu915** [comment]
@AkiKagura Hi, I would like to ask about face idnetity preservation. Does the results and the training data look like the same person? According to my experiments, the results and the training data have similar haircut, skin color, etc. But is totally another person's face.

**Comment by AkiKagura** [comment]
@yuxu915 I think your training steps is not enough. In my experiment, when I trained 1000 steps, it performed like your described, but it performed much better when I train 2000 steps. Also, I don’t suggest using person’s real name as your training token.Something like ‘sks’ is better.

**Comment by yuxu915** [comment]
@AkiKagura thanks for your reply, I set training steps to 2k, and find it's good to generate man portrait(or face reconstruction). However, face is not satisfied when generating images with rich scene or background, is that the limitations of Dreambooth? Some bad cases are as follows (Fig1. "A photo of sks man at Mt. Fuji"; Fig.2 "A photo of sks man swimming in the pool")
![A photo of sks man at Mt  Fuji_final](https://user-images.githubusercontent.com/94051337/202389894-eb4cc4f2-3710-4bdf-94a3-509b38988e0c.png)
:
![A photo of sks man swimming in the pool_final](https://user-images.githubusercontent.com/94051337/202390347-20092305-385f-492b-98d8-8282a46cb079.png)

**Comment by AkiKagura** [comment]
@yuxu915 Yes, that is also the problem I met. To be honest, I’m still having problem like this. I’m planning to try on some human’s distant views this week and I’ll tell you result after that.

**Comment by AkiKagura** [comment]
@yuxu915 Well, I trained on 20 bust and full body portraits, and the generated faces were distorted. Maybe it would be better to use portraits without background? But I don't have that patience🥲.

**Comment by yuxu915** [comment]
@AkiKagura  Maybe it's the inherent limitation of DreamBooth? however, I find results in [this video](https://www.youtube.com/watch?v=Z-hyKADmHmE) seems satisfied. But have not tested on generating image with more rich background.

**Comment by AkiKagura** [comment]
@yuxu915 I tried to remove the background of the training images and added several full body images inside them, but the faces were not so good. Then I removed the full body images, only using the half body to head images for training. That performed better than before, but the faces were still distorted. Maybe it would be better to use only head images? But that may generate bodies unfit the face.

**Comment by yuxu915** [comment]
@AkiKagura If you generate imgs of half-body, distorted faces can be avoided, however the face identity is far from training set, be like 
![huangbo](https://user-images.githubusercontent.com/94051337/203467822-b3d06950-4c0b-4b59-b468-9c1458dea1e0.PNG)
training set and prompt is same as above.

**Comment by nawnie** [comment]
Try taking your completed ckpts. And running them for about 500 more steps under a smaller learning rate.
